components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/utils/SolaceNotifierUtils.java

```diff
@@ -123,6 +125,59 @@ private boolean isSolaceGatewayDetailsAdded() {
         return true;
     }
 
+    /**
+     * Update Async API definition.
+     *
+     * @param api         API to be updated with async API definition in  Solace broker.
+     * @param environment Environment in which the API belongs to.
+     * @throws APIManagementException is error occurs when renaming the application.
+     */
+    public static void updateAsyncAPIDefinition(API api, Environment environment) throws
+            APIManagementException {
+        SolaceAdminApis solaceAdminApis = SolaceNotifierUtils.getSolaceAdminApis();
+        String organization = environment.getAdditionalProperties().get(SolaceConstants
+                .SOLACE_ENVIRONMENT_ORGANIZATION);
+        if (log.isDebugEnabled()) {
+            log.info("Updating Async API definition in Solace...");
+        }
+        String apiNameForRegistration = api.getId().getApiName() + "-" + api.getId().getVersion();
+        String apiDefinition = api.getAsyncApiDefinition();
+        Aai20Document aai20Document = (Aai20Document) Library.readDocumentFromJSONString(apiDefinition);
+        String[] apiContextParts = api.getContext().split("/");
+        String apiNameWithContext = environment.getName() + "-" + api.getId().getName() + "-" + apiContextParts[1] +
+                "-" + apiContextParts[2];
+
+        CloseableHttpResponse response1 = solaceAdminApis.updateAPIDefinition(organization, apiNameForRegistration,
+                apiDefinition);
+        if (response1.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
```

Comment:
According to your current code in updateAPIDefinition, it can return null and this line can cause NPEs

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/11001/commits/2a3e26b73a399ab21f19189d6e430fdeba11e5df